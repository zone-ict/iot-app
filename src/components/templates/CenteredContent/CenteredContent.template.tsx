import React from 'react';
import tw, { styled } from 'twin.macro';

const Container = styled.main`
  ${tw`w-full flex justify-center`}
`;

const Content = styled.div`
  ${tw`w-full sm:max-w-md lg:max-w-xl xl:max-w-2xl flex flex-col`}
`;

interface Props extends React.HTMLAttributes<HTMLDivElement> {
  children: React.ReactNode;
}

export default function CenteredContent({ children, ...props }: Props) {
  return (
    // We're making a generic component that extends a basic div element, so it should be able to accept div props
    // eslint-disable-next-line react/jsx-props-no-spreading
    <Container {...props}>
      <Content>{children}</Content>
    </Container>
  );
}
