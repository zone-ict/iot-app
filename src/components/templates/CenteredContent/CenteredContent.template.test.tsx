import React from 'react';
import renderer from 'react-test-renderer';
import CenteredContent from './CenteredContent.template';

describe('Centered content template', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<CenteredContent>Test</CenteredContent>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
