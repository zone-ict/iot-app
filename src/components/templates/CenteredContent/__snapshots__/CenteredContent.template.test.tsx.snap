// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Centered content template should render correctly 1`] = `
<main
  className="css-1tmw1td"
>
  <div
    className="css-1slw6ji"
  >
    Test
  </div>
</main>
`;
